PokerGame/PokerEngine/Poker.cs

```diff
@@ -115,29 +117,52 @@ private static List<Card> CalculateHandWithHighestCardAndKicker(List<Card> handO
             return null;
         }
 
-        private static bool HasPair(List<Card> hand, out Rank pair)
+        private static Rank CalculateHighestPair(Rank pairOneRank, Rank pairTwoRank)
         {
-            for (var i = 0; i < hand.Count; i++)
+            var hand = new List<Card>()
             {
-                var subHand = hand.GetRange(i + 1, hand.Count - i - 1);
-                foreach (Card card in subHand)
+                new Card() { Rank = pairOneRank },
+                new Card() { Rank = pairTwoRank }
+            };
+            var highestPairRank = HighestCard(hand).Rank;
+            return highestPairRank;
+        }
+
+        private static int NumberOfMostUbundentSameRankedCard(List<Card> hand, out Rank rankOfMostUbendentCard)
+        {
+            var countOfCard = 0;
+            var highestCount = countOfCard;
+            rankOfMostUbendentCard = 0;
+
+            foreach (var cardToCount in hand)
+            {
+                countOfCard = hand.Count(card => card.Rank == cardToCount.Rank);
```

Comment:
add this line first to skip counting again:
```
if (cardToCount.Rank == rankOfMostUbendentCard) 
    continue;
```

Reply:
good idea :)